### PR TITLE
Add client support for drift detection

### PR DESCRIPTION
**Pull request by eranelbaz**
### Issue & Steps to Reproduce / Feature Request
[//]: <> (choose “resolves” “fixes” or “closes” and put link for the issue)
We want to add drift detection support for our terraform provider
for that, we first need to add API client support

### Solution
add client support :)

### CODE REVIEW

**Comment by eranelbaz** [comment]
[Terraform Provider - Drift Detection Deployments](https://trello.com/c/7Ko5En8w/5608-terraform-provider-drift-detection-deployments)

File: client/environment_drift_detection.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	err := self.http.Put("/scheduling/drift-detection/environments/"+environmentId, payload, &result)
+	if err != nil {
+		return EnvironmentSchedulingExpression{}, err
+	}
+
+	return result, nil
+}
+
+func (self *ApiClient) EnvironmentStopDriftDetection(environmentId string) error {
+	err := self.http.Put("/scheduling/drift-detection/environments/"+environmentId, EnvironmentSchedulingExpression{Enabled: false, Cron: ""}, &EnvironmentScheduling{})
```
Comment (RLRabinowitz) [inline client/environment_drift_detection.go:25]: We've discussed this briefly at sync today, @alonnoga 's changing the API to be PATCH in [this PR](https://github.com/env0/env0/pull/6758)

Please use `Patch` here

Comment (RLRabinowitz) [inline client/environment_drift_detection.go:25]: Sorry @eranelbaz but another thing that was just raised. `cron` would now be optional, and `''` wouldn't really work well in the API, as it's a patch

https://env0.slack.com/archives/C02TRUU5YS0/p1646659554487259

Comment (RLRabinowitz) [inline client/environment_drift_detection.go:25]: Sorry @eranelbaz but we're changing `cron` to be optional, and `''` wouldn't really work here
https://env0.slack.com/archives/C02TRUU5YS0/p1646659554487259

Comment (eranelbaz) [reply client/environment_drift_detection.go]: 7b8a0530(convert to Patch)

Comment (eranelbaz) [reply client/environment_drift_detection.go]: same as above

File: client/model.go [hunk 2]
```diff
@@ -393,7 +393,7 @@ type PolicyUpdatePayload struct {
 
 type EnvironmentSchedulingExpression struct {
 	Cron    string `json:"cron"`
-	Enabled bool   `json:"enabled"`
+	Enabled bool   `json:"enabled,omitempty"`
```
Comment (RLRabinowitz) [inline client/model.go:396]: I think the `omitempty` should be on `cron` and not on `enabled`.

A simple test in `model_test.go` could test this behavior

File: client/environment_drift_detection_test.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	. "github.com/env0/terraform-provider-env0/client"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("EnvironmentDriftDetection", func() {
+	environmentId := "env"
+	path := "/scheduling/drift-detection/environments/" + environmentId
+	mockError := errors.New("I don't like milk")
```
Comment (RLRabinowitz) [inline client/environment_drift_detection_test.go:14]: Liar

**Review by RLRabinowitz — APPROVED** [review]
Nice one.
Since we've made some changes ([here](https://github.com/env0/env0/pull/6820/files)), now `enabled` is mandatory and `cron` is not. So you'd need to adjust to model before merging (`omitempty` is on the wrong field)

**Pull request merged**